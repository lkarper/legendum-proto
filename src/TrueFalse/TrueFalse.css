.TrueFalse__container legend {
    font-size: 22px;
}

.TrueFalse__container label {
    font-size: 20px;
}

.TrueFalse__submit {
    margin: 5px;
    font-size: 20px;
}

.TrueFalse__input-div {
    margin: 5px 0;
}

.TrueFalse__input-div label {
    margin: 0 5px;
}
