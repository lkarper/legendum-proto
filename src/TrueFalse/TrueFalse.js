import React, { useState } from 'react';
import PropTypes from 'prop-types';
import QuestionDialogue from '../QuestionDialogue/QuestionDialogue';
import QuestionLegend from '../QuestionLegend/QuestionLegend';
import './TrueFalse.css';

const TrueFalse = (props) => {
    const { 
        page, 
        savedUserInput, 
        checkAnswer,
    } = props;

    const [userChoice, setChoice] = useState();

    if (Object.keys(page).length === 0) {
        return (
            <p>Error: Looks like something went wrong. Check your connection and try again.</p>
        );
    }

    return (
        <div className='TrueFalse__container'>
            <QuestionDialogue 
                page={page} 
                savedUserInput={savedUserInput} 
            />
            <form
                onSubmit={(e) => {
                    checkAnswer(e, userChoice);
                    setChoice();
                }}
            >
                <fieldset>
                    <QuestionLegend 
                        page={page} 
                        savedUserInput={savedUserInput} 
                    />
                    <div
                        className='TrueFalse__input-div'
                    >
                        <input 
                            className='TrueFalse__input radio'
                            type='radio'
                            id='true'
                            name='user-response'
                            value='True/Vērum'
                            checked={userChoice === 'True/Vērum'}
                            onChange={(e) => setChoice(e.target.value)}
                            required
                        />
                        <label htmlFor='true'>True/Vērum</label>
                    </div>
                    <div
                        className='TrueFalse__input-div'
                    >
                        <input 
                            className='TrueFalse__input radio'
                            type='radio'
                            id='false'
                            name='user-response'
                            value='False/Falsum'
                            checked={userChoice === 'False/Falsum'}
                            onChange={(e) => setChoice(e.target.value)}
                            required
                        />
                        <label htmlFor='false'>False/Falsum</label>
                    </div>
                </fieldset>
                <button
                    className='TrueFalse__submit button'
                    type='submit'
                    disabled={!userChoice}
                >
                    Submit
                </button>
            </form>
        </div>
    );
}

TrueFalse.defaultProps = { 
    page: {}, 
    savedUserInput: {}, 
    checkAnswer: () => {},
};

TrueFalse.propTypes = { 
    page: PropTypes.object, 
    savedUserInput: PropTypes.object, 
    checkAnswer: PropTypes.func,
};

export default TrueFalse;
