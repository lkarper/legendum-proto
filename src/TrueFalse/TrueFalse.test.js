import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TrueFalse from './TrueFalse';

describe('TrueFalse component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TrueFalse />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<TrueFalse />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
