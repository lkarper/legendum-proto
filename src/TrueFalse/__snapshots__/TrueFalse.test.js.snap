// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrueFalse component renders the UI as expected 1`] = `
<TrueFalse
  checkAnswer={[Function]}
  page={Object {}}
  savedUserInput={Object {}}
>
  <p>
    Error: Looks like something went wrong. Check your connection and try again.
  </p>
</TrueFalse>
`;
